plugins/GSdx/GSState.cpp

```diff
@@ -405,9 +405,30 @@ GSVideoMode GSState::GetVideoMode()
 	return videomode;
 }
 
-GSVector4i GSState::GetDisplayRect(int i, bool merged_rect)
+GSVector4i GSState::GetDisplayRect(int i)
 {
-	if(i < 0) i = IsEnabled(1) ? 1 : 0;
+	if (!(IsEnabled(0) || IsEnabled(1)))
+		return GSVector4i(0);
+
+	// If no specific context is requested then pass the merged rectangle as return value
+	if (i == -1)
+	{
+		if (m_regs->PMODE.EN1 & m_regs->PMODE.EN2)
+		{
+			GSVector4i r[2] = { GetDisplayRect(0), GetDisplayRect(1) };
+			GSVector4i r_intersect = r[0].rintersect(r[1]);
+			GSVector4i r_union = r[0].runion_ordered(r[1]);
+
+			if (!(r_intersect.width() && r_intersect.height()) ||
```

Comment:
It is fine. But potentially it could be easier to use `?` This way it is easier to see that code will always return. And it avoid the strange indentation on the if.
```
/* the comment here */
bool can_be_merged = bla ..intersect... bla
return can_be_merged ? r_union : r[...area...];
```